### PR TITLE
Update to modern syntax

### PR DESCRIPTION
**Pull request by didriklundberg**

### CODE REVIEW

**Review by palmskog — APPROVED** [review]
@didriklundberg this looks good to me, but "local theorems" / trivialities are not converted. Will you convert `prove("xyz", ... )` to `Theorem xyz[local]: ...` in these files in a future PR?

**Comment by didriklundberg** [comment]
> @didriklundberg this looks good to me, but "local theorems" / trivialities are not converted. Will you convert `prove("xyz", ... )` to `Theorem xyz[local]: ...` in these files in a future PR?

Good suggestion. I was planning to add more into this PR, depending on your feedback. I'll see how much work it is.

**Review by palmskog — DISMISSED** [review]
Fine by me, but note that there will be merge work required for #174. Which order do you think we should merge? I would personally prefer for #174 to go in first.

Note that it's not feasible to manually review all the changes, we have to go by CI and sampling.

**Comment by arolle** [comment]
Looks good to me. Your automatic conversion covers the vast majority of occurrences. There are a few remaining ones, e.g. occurrences of `prove` in `*Script.sml` files that can be fixed by hand later.

This is ready to be merged, however as mentioned elsewhere there can be a couple of merge conflicts in the near future due to these changes. Any branches that remain to be merged into master should probably go before this PR, to simplify their merge.

**Review by arolle — APPROVED** [review]

**Comment by arolle** [comment]
In case you want to further improve your script (no need to!), here are some of the occurrences that seem to remain:
```
grep --recursive --include='*Script.sml' 'val.*=.*\(\<save_thm\>\|\<prove\>\)'
```

**Review by palmskog — DISMISSED** [review]
Approving again.

File: src/theory/tools/backlifter/bir_arm8_backlifterScript.sml [hunk 1]
```diff
@@ -134,9 +137,26 @@ Q.SUBGOAL_THEN
...
+EVAL_TAC
+QED
+
 
-val arm8_vars_def = Define `
+Theorem imm_eq_to_val_eq[local]:
+  !a b . ((BVal_Imm(Imm1 a)) = (BVal_Imm(Imm1 b))) = (a = b)
+Proof
+REPEAT STRIP_TAC >> EVAL_TAC
+QED
```
Comment (palmskog) [inline src/theory/tools/backlifter/bir_arm8_backlifterScript.sml:157]:
```suggestion
```
This is an unused local lemma that I deleted in #174. It's basically dead code.

File: src/theory/tools/backlifter/bir_arm8_backlifterScript.sml [hunk 2]
```diff
@@ -134,9 +137,26 @@ Q.SUBGOAL_THEN
...
+
+Theorem bool2w_and[local]:
+  !a b. ((bool2w a) && (bool2w b)) = (bool2w (a /\ b))
+Proof
+REPEAT STRIP_TAC >>
+FULL_SIMP_TAC std_ss [bool2w_def] >>
+Cases_on `a` >>
+Cases_on `b` >>
+EVAL_TAC
+QED
```
Comment (palmskog) [inline src/theory/tools/backlifter/bir_arm8_backlifterScript.sml:150]:
```suggestion
```
This is an unused local lemma I deleted in #174. It's basically dead code.

File: src/theory/tools/backlifter/bir_arm8_backlifterScript.sml [hunk 3]
```diff
@@ -375,24 +397,98 @@ EVAL_TAC  >>
...
+    bmr_rel r bs' ms'
+Proof
+REPEAT STRIP_TAC >>
+ASSUME_TAC (ISPECL [``r:(64, 8, 'b) bir_lifting_machine_rec_t``, ``mu:64 word_interval_t``,
+                    ``mms:(word64# word8 list) list``,
+                    ``p:'a bir_program_t``] bir_is_lifted_prog_MULTI_STEP_EXEC) >>
+REV_FULL_SIMP_TAC std_ss [] >>
+QSPECL_X_ASSUM ``!n ms bs. _`` [`n'`, `ms`, `bs`] >>
+REV_FULL_SIMP_TAC (std_ss++holBACore_ss) [bir_state_is_terminated_def]
+QED
```
Comment (palmskog) [inline src/theory/tools/backlifter/bir_arm8_backlifterScript.sml:473]:
```suggestion
```
These lemmas were moved to `bir_common_backlifterScript.sml` in #174.

File: src/theory/tools/backlifter/bir_arm8_backlifterScript.sml [hunk 4]
```diff
@@ -464,10 +561,62 @@ subgoal `ms'.PC IN mls` >- (
...
+ * But PC has index 0... *)
+subgoal `bs''.bst_pc.bpc_index = 0` >- (
+  METIS_TAC [bir_exec_to_addr_label_n_ended_running, bir_state_is_terminated_def]
+) >>
+(* ... which proves the goal, after some identification of states. *)
+FULL_SIMP_TAC std_ss [bir_state_COUNT_PC_def, bir_exec_to_addr_label_n_def,
+		      bir_exec_to_labels_n_def,
+		      bir_exec_steps_GEN_SOME_EQ_Ended] >>
+REV_FULL_SIMP_TAC (std_ss++holBACore_ss) [bir_state_is_terminated_def]
+QED
```
Comment (palmskog) [inline src/theory/tools/backlifter/bir_arm8_backlifterScript.sml:615]:
```suggestion
```
This was moved to `bir_common_backlifterScript.sml` in #174

**Review by palmskog — APPROVED** [review]
I noticed some removed lemmas made a comeback...

**Comment by didriklundberg** [comment]
> I noticed some removed lemmas made a comeback...

My apologies, this must be from the rebase I did after merging #174

**Review by palmskog — APPROVED** [review]

**Pull request merged**